databroker/src/grpc/kuksa_val_v1/val.rs

```
@@ -344,12 +283,146 @@ impl proto::val_server::Val for broker::DataBroker {
+
+        // Copy (to move into task below)
+        let broker = self.clone();
+
+        // Create stream (to be returned)
+        let (sender, receiver) = mpsc::channel(10);
+        // Listening on stream
+        tokio::spawn(async move {
+            let permissions = permissions;
+            let broker = broker.authorized_access(&permissions);
+            loop {
+                select! {
+                    message = stream.message() => {
+                        match message {
+                            Ok(request) => {
+                                match request {
+                                    Some(req) => {
+                                        let entry_updates = req.updates;
+
+                                        // Collect errors encountered
+                                        let mut errors = Vec::<DataEntryError>::new();
+                                        let mut updates = Vec::<(i32, broker::EntryUpdate)>::new();
+
+                                        for request in entry_updates {
+                                            match &request.entry {
+                                                Some(entry) => match broker.get_id_by_path(&entry.path).await {
+                                                    Some(id) => {
+                                                        match validate_entry_update(&broker, &request, id).await {
+                                                            Ok(pair) => {
+                                                                updates.push((pair.first, pair.second));
+                                                            }
+                                                            Err(e) => {
+                                                                let message = format!("Data present in the request is invalid: {}", e.message());
+                                                                errors.push(proto::DataEntryError {
```

Some errors which previously early returning the function (see here: https://github.com/eclipse-kuksa/kuksa-databroker/pull/34/files#diff-764f0adb7e58cb82e14ca30770f38dce1b3a6d003d729ddd651442ece69e1509L235) were converted to DataEntryError
While for a synchronous set we want to fail fast, for the async streaming api we don't want to end the loop in case an error occurs due to an invalid data point.